src/transformers/training_args.py

```diff
@@ -510,10 +522,19 @@ def __post_init__(self):
             self.output_dir = os.getenv("SM_OUTPUT_DATA_DIR")
         if self.disable_tqdm is None:
             self.disable_tqdm = logger.getEffectiveLevel() > logging.WARN
-        self.evaluation_strategy = EvaluationStrategy(self.evaluation_strategy)
-        self.logging_strategy = LoggingStrategy(self.logging_strategy)
+
+        if isinstance(self.evaluation_strategy, EvaluationStrategy):
+            warnings.warn(
+                "using `EvaluationStrategy` for `evaluation_strategy` is deprecated and will be removed in version 5 of 🤗 Transformers. Use `IntervalStrategy` instead",
+                FutureWarning,
+            )
```

Comment:
This will only detect if an enum value is passed to the Trainer. Ideally we want to throw the warning as soon as someone uses `EvaluationStrategy`, though that's a bit harder. I can look at it after this PR is merged if you prefer.